log/log.go

```diff
@@ -64,25 +64,33 @@ var bufStdErr bytes.Buffer
 // that can be used for checking what the logger would've written. This is
 // mostly used for tests.
 func OutputToBuf() {
+	debugMut.Lock()
+	defer debugMut.Unlock()
 	stdOut = &bufStdOut
 	stdErr = &bufStdErr
 }
 
 // OutputToOs redirects the output of the log.*-outputs again to the os.
 func OutputToOs() {
+	debugMut.Lock()
+	defer debugMut.Unlock()
 	stdOut = os.Stdout
 	stdErr = os.Stderr
 }
 
 // GetStdOut returns all log.*-outputs to StdOut since the last call.
 func GetStdOut() string {
+	debugMut.Lock()
+	defer debugMut.Unlock()
 	ret := bufStdOut.String()
 	bufStdOut.Reset()
 	return ret
 }
 
 // GetStdErr returns all log.*-outputs to StdErr since the last call.
 func GetStdErr() string {
```

Comment:
Sorry One more final thought: The idea is to be able to detect if an error message or a general message has been logged in case of a specific event right ?
Would it be difficult to provide some kind of helper method in order to do that ? `log.OutputContains(text)`  which returns true if the text is contained in the log (i.e. when *it is buffered*) or false if it's not OR if the log is not directed to the buffer.
What do you think ?
Otherwise I'm OK to merge after this discussion.